secrets/azurekeyvault/akv.go

```diff
@@ -208,7 +208,7 @@ func dial(useCLI bool) (*keyvault.BaseClient, error) {
 
 var (
 	// Note that the last binding may be just a key, or key/version.
-	keyIDRE = regexp.MustCompile("^(https://.+\\.vault\\.azure\\.net/)keys/(.+)$")
+	keyIDRE = regexp.MustCompile("^(https://.+\\.vault\\.(?:azure\\.net|azure\\.cn|usgovcloudapi\\.net|microsoftazure\\.de)/)keys/(.+)$")
```

Comment:
Use *raw strings* for regular expressions to avoid excessive quoting.

```suggestion
	keyIDRE = regexp.MustCompile(`^(https://.+\.vault\.(?:azure\.net|azure\.cn|usgovcloudapi\.net|microsoftazure\.de)/)keys/(.+)$`)
```

And you might want to prefix your commit with `secrets/azurekeyvault:` (the PR might get discarded if the commit is merged with `ff`).

Reply:
@wmark I didn't use the raw string to stick with the current convention, would you prefer me to switch it to the in this case?

Reply:
As suggested, I've updated this to use a raw string and squashed the commit messages